### DISCUSSION TITLE
Getting "The response type is not supported." on subscription request

### DISCUSSION THREAD

**Discussion by peterhuijsen** [ChilliCream/hotchocolate#5045] [Q&A]
When I'm doing normal queries to my gql api and mutations it works as expected, but when creating a subscription it throws me this error:

`{
	"errors": [
		{
			"message": "The response type is not supported."
		}
	]
}`

This is my setup file:

```cs
var builder = WebApplication.CreateBuilder(args);

builder.Services
    .AddGraphQLServer()
    .AddQueryType<Query>()
    .AddMutationType<Mutation>()
    .AddSubscriptionType<Subscription>()
    .AddMongoDbFiltering()
    .AddMongoDbSorting()
    .AddInMemorySubscriptions();

builder.Services
    .AddCors(options =>
    {
        options.AddDefaultPolicy(cors => 
            cors.AllowAnyHeader()
                .AllowAnyMethod()
                .AllowAnyOrigin()
        );
    });

var app = builder.Build();
app.UseHttpsRedirection();
app.UseRouting();
app.UseCors();
app.UseWebSockets();
app.MapGraphQL();
app.Run();
```

and this is the subscription:

```cs
public class Subscription
{
    [Subscribe]
    public Note NoteUpdated([EventMessage] Note note) => note;
}
```

with the object:

```cs
public class Note
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; }
    public string Title { get; set; }
    public string? Content { get; set; }
    public IReadOnlyList<Tag>? Tags { get; set; }
}
```

and the request I tried:

```gql
subscription OnNoteUpdated {
  noteUpdated {
    id
    title
    content
    tags {
      title
    }
  }
}
```

I have tried using other `Content-Type` headers, but this made it return a 404 error.
I'm running this locally and I'm using Insomnia to send the queries

I feel like I'm just missing something pretty obvious, but I can't find it, so I didn't make this an issue and instead a discussion. Does anyone else have any ideas?

**Comment by PascalSenn** [comment]
i dont see anything obvious wrong. do you have a reproduction on github?

**Comment by bfeucht-wof** [comment]
Did you ever resolve this? I am currently facing the same issue. My subscriptions had previously been working fine, but they have all of a sudden stopped working.

**Reply by peterhuijsen** [reply]
Nope, ended up using another gql client (apollo/banana cake pop) which did work.
Still experiencing this error on insomnia (and others like hoppscotch) unfortunately.

It seems like it has something to do with the way subscriptions are implemented in hot-chocolate (apollo pub-sub model), but I might be completely wrong.
